Graphviz/DOT.sublime-syntax

```diff
@@ -1,62 +1,391 @@
+        8: punctuation.definition.string.end.dot
+      push: expect-declaration-statements-list-body
+
+    - match: (?=\{)
+      push: expect-declaration-statements-list-body
+
+    - include: stray-braces
+
+###[ DECLARATIONS / STATEMENTS ]###############################################
+
+  node-statements:
+    # NOTE: compass point values are not keywords
+
+    - match: |-
+        (?x:
+          (?:\b({{id}})\b\s*)
+          (?:(:)\s*\b({{id}})\b\s*)?
+          (?:(:)\s*\b({{compass_pt}})\b\s*)?
+          (\[)
+        )
+      captures:
+        1: entity.name.node.dot
+        2: punctuation.separator.dot
+        3: entity.name.node.dot
+        4: punctuation.separator.dot
+        5: meta.annotation.dot variable.annotation.compass.dot
+        6: punctuation.section.brackets.begin.dot
+      push: attribute-list-body
+
+  attribute-statements:
+    - match: |-
+        (?x:
+          (?:\b({{graph}}|{{node}}|{{edge}})\b\s*)
+          (\[)
+        )
+      captures:
+        1: keyword.declaration.dot
+        2: punctuation.section.brackets.begin.dot
+      push: attribute-list-body
+
+###[ LITERALS ]################################################################
+
+  literals:
+    - include: literal-strings
+    - include: literal-constants
+    - include: literal-numbers
+
+  literal-constants:
+    - match: \b(?:false)\b
+      scope: constant.language.boolean.false.dot
+    - match: \b(?:true)\b
+      scope: constant.language.boolean.true.dot
+
+    - match: \b(?:{{svg_colors}}|{{x11_colors}}|{{brewer_color_schemes}})\b
+      scope: constant.language.color.dot
+
+    - match: |-
+        (?x:
+          \b(?:
+            {{layout_engines}}
+          | {{graph_attributes}}
+          | {{subgraph_attributes}}
+          | {{cluster_attributes}}
+          | {{node_attributes}}
+          | {{edge_attributes}}
+          | {{arrow_shape_attributes}}
+          | {{polygon_shape_attributes}}
+          | {{record_shape_attributes}}
+          | {{ranksep_attributes}}
+          | {{splines_attributes}}
+          )\b
+        )
+      scope: support.constant.attribute.dot
+
+    - match: '\b(?:{{id}})\b'
+      scope: entity.name.node.dot
+
+  literal-numbers:
+    - match: |-
+        (?x:
+          (?:(?:([-])|([+]))(\s+)??)?
+          (
+            (?: (0+)?? [1-9] \d* | 0 )
+            (?: (\.) \d* )                                                      # 1. 1.1
+          | (\.) \d+                                                            # .1
+          )
+        )
+      scope: meta.number.float.decimal.dot
+      captures:
+        1: constant.numeric.sign.dot
+        2: invalid.illegal.numeric-sign.dot
+        3: invalid.illegal.whitespace.dot
+        4: constant.numeric.value.dot
+        5: invalid.illegal.leading-zero.dot
+        6: punctuation.separator.decimal.dot
+        7: punctuation.separator.decimal.dot
+
+    - match: |-
+        (?x:
+          (?: (?:([-])|([+])) (\s+)?? )?
```

Comment:
What's the reason for `(\s+)??` Why not simply `(\s*)` or `(\s*?)` ?

```suggestion
          (?: (?:([-])|([+])) (\s*) )?
```

Maybe syntax tests could be added to justify/verify certain special pattern usages?